File: test/testload.pl
#!/usr/bin/env perl
use v5.14;

use Benchmark qw(timeit timestr);

sub stuff {
	my $start;
	my @flights;

	while (<>) {
		chomp;
		unless ($start) {
			$start = $_;
			next;
		}

		my ($from, $to, $day, $price) = split /\s+/;
		push @flights, {from => $from, to => $to, day => $day, price => $price};
	}

	say($start);
	say(scalar @flights);
}

say(timestr(timeit(1, 'stuff')));

File: testload.rb
#!/usr/bin/env ruby
# -*- encoding: utf-8 -*-

require 'benchmark'

Flight = Struct.new(:from, :to, :day, :price)

def parse(filename)
  start = nil
  flights = []

  File.readlines(filename).each do |line|
    if start.nil?
      start = line.chomp
      next
    end

    flights << Flight.new(*line.split(' '))
  end

  {:start => start, :flights => flights}
end

puts Benchmark.measure {
  ret = parse(ARGV[0] || '/dev/stdin')

  puts ret[:start]
  puts ret[:flights].length
}
